test/unit/test_cache.py

```diff
@@ -461,81 +474,80 @@ def test_load_with_expired_entries(self, tmpdir):
 
         cache_path = os.path.join(tmpdir, "cache2.txt")
-        c1 = cache.SFDictFileCache(file_path=cache_path)
+        c1 = NeverSaveSFDictFileCache(file_path=cache_path)
         c1["key"] = BrokenWritingPickleObject()
-        assert not c1._save()
+        assert not c1.save()
         assert not os.path.exists(cache_path)
 
-        c2 = cache.SFDictFileCache(file_path=cache_path)
-        assert not c2._load()  # load should return false due to no file
+        c2 = NeverSaveSFDictFileCache(file_path=cache_path)
+        assert not c2.load()  # load should return false due to no file
 
 
 def test_file_is_not_updated(tmpdir):
     tmp_cache_file = os.path.join(tmpdir, "tmp_cache")
-    sfcache = cache.SFDictFileCache(file_path=tmp_cache_file, entry_lifetime=1)
+    sfcache = NeverSaveSFDictFileCache(file_path=tmp_cache_file, entry_lifetime=1)
     sfcache.update({"key": "value"})
     assert sfcache._cache_modified
-    sfcache._save()  # this save call will dump cache to file because item was added
+    sfcache.save()  # this save call will dump cache to file because item was added
     assert not sfcache._cache_modified
     updated_time = os.path.getmtime(tmp_cache_file)
-    sfcache._save()  # this save call will be a no-op since there is no update
+    sfcache.save()  # this save call will be a no-op since there is no update
     assert os.path.getmtime(tmp_cache_file) == updated_time
     sfcache.update({"key": "value2"})
     assert sfcache._cache_modified
     time.sleep(0.1)  # sleep 0.1 to avoid flushing too fast
-    sfcache._save()  # this save call will dump cache to file
+    sfcache.save()  # this save call will dump cache to file
     assert not sfcache._cache_modified
     second_updated_time = os.path.getmtime(tmp_cache_file)
     assert second_updated_time > updated_time
     assert sfcache["key"] == "value2"
     assert not sfcache._cache_modified
-    sfcache._save()  # this save call will be a no-op since there is no update
+    sfcache.save()  # this save call will be a no-op since there is no update
     assert not sfcache._cache_modified
     assert os.path.getmtime(tmp_cache_file) == second_updated_time
     time.sleep(1)
-    sfcache._save()  # this save call will dump cache because cache item is expired
+    sfcache.save()  # this save call will dump cache because cache item is expired
     assert not sfcache._cache_modified
     assert os.path.getmtime(tmp_cache_file) > second_updated_time
 
-    cache3 = cache.SFDictFileCache(file_path=tmp_cache_file, entry_lifetime=1)
+    cache3 = NeverSaveSFDictFileCache(file_path=tmp_cache_file, entry_lifetime=1)
     assert len(cache3) == 0
     os.unlink(tmp_cache_file)
 
 
 def test_cache_do_not_write_while_set_item(tmpdir):
     tmp_cache_file = os.path.join(tmpdir, "tmp_cache")
-    sfcache = cache.SFDictFileCache(tmp_cache_file)
+    sfcache = NeverSaveSFDictFileCache(tmp_cache_file)
```

Comment:
same revert

Reply:
I would like to leave it like this, as this makes the test deterministic instead of leaving things up to Python's random number generator